### ISSUE TITLE
Difficult to delete a backup job using screen reader

### ISSUE THREAD

**Issue by citygabor** [duplicati/duplicati#5098]
- [x] I have searched open and closed issues for duplicates.
- [x] I have searched the [forum](https://forum.duplicati.com) for related topics.

----------------------------------------

## Environment info

Local
- **Duplicati version**: 2.0.7.1_beta_2023-05-25
- - **Operating system**: TrueNas(docker image)
- **Backend**: TrueNAS

## Description

Hello Duplicati Team,

I would like to use your great product, but when I noticed I cannot delete my test backups with my screen reader.

When I am on that to delete any created backup job I cannot read the picture with screen reader. So I would like to suggest to change that picture question to the backup job name. As it is not an accessibility friendly concept to ask this from who has visual impairment problems.

Just like in TrueNAS scale tool whenever I want to delete a dataset it is required to type the mount path name and after I typed it correctly I could delete the dataset.

Please change it as it would be may be a better choice as this picture choosing option it is not related to a backupat all.

Thanks,

Regards,

Gabor

## Steps to reproduce

1. When I created a backup job and finished the backup successfuly.
2. I go to the Home.
3. Select the related job.
4. Click on the delete tag.
5. Selecting the checkbox: "Delete remote files "
6. Then hit the "Delete backup " button.
7. Facing the picture question.(attached the screen)

- **Actual result**:

- **Expected result**:
To easily delete backup job without a picture question or implement as a backup name for the backup jobs. Which can be easily enterred with even with screen reader.

## Screenshots

![DuplicatiDeleteJob](https://github.com/duplicati/duplicati/assets/24994660/ee28a6f6-e805-4385-bb7d-c0ea6c7bcf39)
![DuplicatiDeleteJob](https://github.com/duplicati/duplicati/assets/24994660/e697f8f0-41d7-471c-ae66-d6c479e33e09)

**Comment by gpatel-fr** [comment]
Hello

The captcha is a good feature to protect against unwanted delete, but your use case has not been envisioned.

There is a way to delete all files using the 'Delete' command in the command line tool, but it's not exactly convenient (it is needed  to remove the parameters, to set the allow-full-removal option, and enter the 'version' option with a number of version sufficient to cover all possibilities, such as 0-10000 for example). I admit it is not really practical.

A good solution would be to allow for the captcha to be played on the computer speaker, but it's not easy to plan for it with the limited resources of this project. In short, I have absolutely no idea on how to do it.

Short of that, would adding an option to disable the captcha confirmation be enough ?

**Comment by citygabor** [comment]
Hello,

 

If that option is not possible easily what I mentioned in my thread ?

To rewrite the delete backup process to request only the backup job name instead of the captcha picture?

 

If this is not possible then yes. 

“would adding an option to disable the captcha confirmation be enough ?

“ This would be a very useful in the settings or in the Backup job option itself too.

any of the listed option would be great to see it.

 

Many Thanks for your help,

regards,

Gabor

**Comment by ts678** [comment]
> Selecting the checkbox: "Delete remote files "

The one Immediate workaround that I can think of is to uncheck that.
Many destinations provide tools allowing quick manual folder deletes.

Remote files hopefully can be deleted in a way not needing CAPTCHA.
If remote is at  a major service, maybe their accessibility work is better.

[advanced deleting of backup configs #1201](https://github.com/duplicati/duplicati/issues/1201) explains the wish to avoid

> ransomware can simply make a http request to the local webserver and this will cause Duplicati to erase the remote backup.

and at the end is a poster (with a [forum topic](https://forum.duplicati.com/t/captcha-eye-roll-also-it-dont-work/13244) and a [GitHub issue](https://github.com/duplicati/duplicati/issues/4625)) who likely doesn't know API ways, but dislikes the CAPTCHA.

[Feature/add delete dialog (#2008)](https://github.com/duplicati/duplicati/commit/17e2a3dab4b7b578b19176e0b963a934c8725893) might be the actual commit for this:

> Added a delete page with options to delete the local database, as well as the remote files.
Added a captcha to protect against automated attacks that would attempt to delete the remote files.

which to me has to resist attacks at both the visible GUI level, and also at the API that the GUI uses to actually get server work.
Current CAPTCHA situation is that the battle has escalated past distorted text, and paid human CAPTCHA solving services exist.

[Added protection against XSRF attacks](https://github.com/duplicati/duplicati/commit/7e0f9860b5de9d524a84dd2874d92ece69b2fe62) which predates the 2016 http note above likely makes attacks a little harder to do right.

Posting this to mention a possible immediate workaround. Beyond that, there's the update on history and the current situation.

**Comment by Jojo-1000** [comment]
I guess for password protected backups you could also request to enter the backup password instead, or confirm the server password (that can be bypassed by logging in from the tray icon). Not sure what to do for unprotected situations. I guess in that case it isn't that sensitive (only local access from same computer and only a single user on that computer), so a simple question would be fine.

"Real" captcha implementations have an audio option for a reason.

By the way @citygabor, it would be interesting to hear feedback on the accessibility of the UI. I can't imagine this is the only situation that is lacking.

**Comment by citygabor** [comment]
Hello,

 

Any captcha never worked for me at the first time and I had to rely on sighted person help. That was the reason I turned to you guys to make it as typing the backup name instead of the captcha or even to turn it off. But it is a bit worried any API calls can  delete my backups.

 

But even the suggested idea can be implemented to typing the admin password also would be a great idea.

 

Regarding the lack of usability I would not say it has mission impossible  to navigate, but I was going to raise a “couple” of suggestions to adding real buttons to the some of the edit, delete, Export, configuration, etc. It would be also great to see that every backup task name tagged as heading 2 or heading 3 html tag  element as it helps in our navigation a lot.

Question 1:

 

Till now I have not discovered yet the pause or abort button or a stop button for a running backup job. Is it possible?

 

 

Question 2:

 

 

Is there any way to specify a folder for my backup inside my /backup  directory?  To keep my backup files in the relevant backup directory?

I could discovered a advanced feature called “Prefix backup name”. I believe this is the purpose to name the file names based on the backups. As later will be a big mess if every backup files are just called as Duplicati-4335rhfh385232475*.zip  Is there any folder creation feature is in this Advanced feature menu or it is not possible? Example: my Backup name: BackupTask1  and My Backup files location would be: /backup/BackupTask1/

 

Thanks,

Regards,

Gabor

**Comment by Jojo-1000** [comment]
Thanks for your feedback @citygabor.

> Till now I have not discovered yet the pause or abort button or a stop button for a running backup job. Is it possible?

There are buttons next to the current task in the header, but they are not focusable and can only be used by mouse.

> Is there any way to specify a folder for my backup inside my /backup  directory?  To keep my backup files in the relevant backup directory?
>
> I could discovered a advanced feature called “Prefix backup name”. I believe this is the purpose to name the file names based on the backups.

Normally you don't need to change the prefix, unless you need multiple backups in the same folder. But there is a lot that can go wrong, so it's better to have it separate like in your example.

On the Backup destination page, you can edit the path as text (for the file destination). There you have to select "Enter path" and then tab back to access the text field. The path for directories should always end in a slash. If you then run a connection test, it should ask to create the folder for you (the tab order for dialogs is also screwed up).

**Comment by gpatel-fr** [comment]
@citygabor 

>> Is there any folder creation feature is in this Advanced feature menu or it is not possible? Example: my Backup name: BackupTask1 and My Backup files location would be: /backup/BackupTask1/

It's quite possible but there is no automaticity. That is, there is no feature that can say 'take the backup name and create automatically a folder on the backend to store the backup files'. However, if the backup name is compatible with the rules for names on the specific backend, it's usually possible to specify a target directory on the backend in the backend configuration, and then the picked name can be made the same as the backup name.
So if your main folder on your backend is /backups, you could have a /backups/backupdriveC and a /backups/backupdriveD with 2 different jobs, one for backup of the C drive, and one for backup of the D drive.

**Comment by citygabor** [comment]
Hello,

 

Thanks for your advise for my both case.

 

1.	Pause button

I can confirm that this is really cannot read by JAWS screen reader. When I pressing my space or enter key around the status on the actual running backup job it tries to enter in a kind of application “mode”, but it immediately exits from that mode. I even tried to use JAWS Cursor, but it does not recognize the pause button either.

 

2.	Folder creation.

I tried earlier to use that destination like: /backup/backuptask1 without “/” (slash). But did not work and left it as it is. But now tried it with the close “/” (slash( and it worked as you explained.

Many thanks,

 

Just confirm one thing regarding the deletion. If I can delete the backup job itself from the Duplicati UI and I delete the backup zip files from my backup NAS which are inside my directory like /backup/backuptask1/ I will not leave any DB or any logs behind the Duplicati, right?

 

As I have just wanted to see clearly what I am going to delete, but with the backup job destination backup directory and even with the backup name prefix I can manage it by own.

 

If you say so, now the                                                                                                                              captcha remote file deletion will not bothers me at all as a workaround can be applied.

 

Thanks for your help,

Regards,

Gabor

**Comment by kenkendk** [comment]
I think this use case is valid, but as mentioned it was not considered initially.
The reason for the Captcha is to protect against rouge scripts, seeing that the communication is via HTTP, a broken sandbox in a browser could allow a website to request a full delete.

My suggestion for this would be to add a commandline argument to the trayicon, say `--no-captcha` which disables the feature. The local installation can then be modified to support screen readers, by changing the shortcut parameters, and this cannot be changed via any HTTP requests. 

For protecting against accidental deletes, the captcha response could be shown as plain text (readable by the screen reader), and the user still needs to enter it. In this configuration, the captcha would not guard against malicious scripts, as they can obtain the captcha token easily.

**Comment by kenkendk** [comment]
The option to disable the captcha is added with #5427 
The option name is `--webservice-disable-visual-captcha`.

The setting is persisted between restarts, so if it is disabled, starting without the option will keep it disabled.
Use `--webservice-disable-visual-captcha=false` to re-enable it if needed.

**Comment by citygabor** [comment]
Hello,

 

Now, I am able to delete the backup jobs and also the content of the Backup jobs.

 

Thanks so much,

Regards,

G
